engine.py

```diff
@@ -912,3 +912,58 @@ def __setup_favorite_dirs(self):
                                 type=(nuke.IMAGE | nuke.SCRIPT | nuke.GEO),
                                 icon=icon_path,
                                 tooltip=path)
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self):
+        """
+        Returns the path to the current file if it resides on
+        disk. If unsaved, it returns None.
+
+        :return: Path to the current scene if file is saved, else returns
+                 and empty string.
+        """
+        # retrieve the name of the nuke file. In case of a saved file, the
+        # absolute path of the file is returned, for an unsaved file,
+        # an empty string is returned.
+        return nuke.root().knob("name").value()
+
+    def get_session_dependencies(self):
+        """
+         Returns the list of file or folder dependencies for the current
+         session.
+
+         :return: A list of file or folder paths that the current
+                  session depends on.
+         """
+        dependencies = []
+
+        # we shall query for file dependencies for the file frame range
```

Comment:
This seems akin to something like a frustum culling in Maya. I am not sure this is always the behaviour we want. I think the default behaviour should be to return *all* dependencies in the scene, regardless of frame range. @rupalkhilari, did you have a particular use case in mind for the frame range "culling"? I think for a basic, cross-DCC and generic implementation, we want to return all dependencies in the scene. Potentially in the future we can add options to limit the set of returned items. What do you think?

Reply:
@manneohrstrom I see your point. My only thought for using the minimum set was that the artist shouldn't be waiting on transfer of data that isn't going to be used for the render. The `nuke.Node.fileDependencies()` also requires a start and end frame. But I see your point - it is possible that frame range could be changed, and at that point it might become tricky then.

Reply:
maybe we could just grab all write and writegeo nodes out of the scene? what does `nuke.Node.fileDependencies()` do?

Reply:
Oh, did you mean Read and ReadGeo nodes ? The nuke.Node.fileDependencies() returns all file dependencies from the node graph traversing upwards from that Node. (more info [here](https://docs.thefoundry.co.uk/nuke/110/pythonreference/nuke.Node-class.html#fileDependencies) for a given frame range. That probably makes sense for rendering, but probably not so good for someone to be working on the file.

Reply:
sorry, yes i meant read and readgeo!